olp-cpp-sdk-core/src/client/PendingUrlRequests.h

```diff
@@ -90,6 +90,13 @@ class PendingUrlRequest {
   /// Will be called when the Network response arrives.
   void OnRequestCompleted(HttpResponse response);
 
+  /// Returns request's status. It is 'true' for merged requests or for requests
+  /// that are waiting for the Network callback
+  bool IsPending() const {
+    std::lock_guard<std::mutex> lock(mutex_);
+    return http_request_id_ != kInvalidRequestId || callbacks_.size() > 1u;
```

Comment:
Ahh, sorry. Missed that you actually check > 1. I thought you only check that there is something to wait for.